### DISCUSSION TITLE
UserControl Shows Blank

### DISCUSSION THREAD

**Discussion by mysteryx93** [AvaloniaUI/Avalonia#10107] [Q&A]
Resuming work on [MvvmDialogs](https://github.com/mysteryx93/HanumanInstitute.MvvmDialogs/). Running in desktop in single-page mode like on mobile devices.

Now I'm seeing something strange, and wanted your input. I can navigate forward fine and show a new View. To navigate back, I'm keeping a weak cache of Views to re-use them if possible; but then they show completely blank?

At this line
https://github.com/mysteryx93/HanumanInstitute.MvvmDialogs/blob/master/src/MvvmDialogs.Avalonia/Navigation/ViewCache.cs#L25

If I re-create a fresh new View every time instead of re-using it from the cache... sometimes it works, sometimes it doesn't. I can now navigate back to MainView (need to click Close in current demo before Show become enabled), I can show the dialog once, navigate back... now it randomly either appears or becomes blank. So I can navigate back once or twice depending on the run before it goes blank. That's without relying on the cache and using fresh new Views.

Any idea what could be causing this?

If anyone wants to try the demo, run "Demo.CrossPlatform.Desktop".

I'm also noting that animations do not work at all. The image freezes until the animation delay elapsed.

    <TransitioningContentControl Content="{Binding CurrentView}">
        <TransitioningContentControl.PageTransition>
            <PageSlide Orientation="Horizontal" Duration="0:00:00.500" />
        </TransitioningContentControl.PageTransition>
    </TransitioningContentControl>

EDIT: If I remove `TransitioningContentControl`, everything works as expected. It's the animation control that is completely broken.

**Comment by maxkatz6** [comment]
Answered here https://github.com/AvaloniaUI/Avalonia/issues/10108#issuecomment-1407223998
